simphony/tools/lattice_tools.py

```diff
@@ -0,0 +1,990 @@
+    ''' Test if primitive vectors describe a body centered orthorhombic
+    lattice
+
+    Also returns True for vectors describing a body centered cubic or
+    a body centered tetragonal lattice
+
+    Numerical errors are tolerated within ``tolerance``
+
+    Parameters
+    ----------
+    p1, p2, p3: array_like
+        primitive vectors
+    tolerance: float
+        tolerance for numerical errors
+
+    Returns
+    -------
+    output : bool
+    '''
+    # lengths of the vectors
+    lenA, lenB, lenC = map(vector_len, (p1, p2, p3))
+
+    if (numpy.isnan((lenA, lenB, lenC)).any() or
+            numpy.isinf((lenA, lenB, lenC)).any()):
+        raise ValueError("Vectors contain invalid values")
+
+    if numpy.less_equal((lenA, lenB, lenC), tolerance).any():
+        raise ValueError("Edge lengths must be strictly positive")
+
+    all_length_equal = (numpy.abs(lenA-lenC) < tolerance and
+                        numpy.abs(lenB-lenC) < tolerance)
+
+    # dot products of vectors
+    dot_products = numpy.abs(map(numpy.dot, (p1, p2, p3), (p2, p3, p1)))
+    right_angles = dot_products <= tolerance
+    one_right_angle = right_angles.sum() == 1
+
+    if one_right_angle:
+        # type 1: as given by the PrimitiveCell class method
+        # p1: (a, 0, 0), p2: (0, b, 0), p3: (a/2, b/2, c/2)
+        # mapping for the lengths of vectors that make right angle
+        # the nonzero dot products are a**2./2 and b**2./2 respectively
+        nonzero_dot = dot_products[~right_angles]
+
+        # find a and b
+        mapping = {0: (lenA, lenB), 1: (lenB, lenC), 2: (lenC, lenA)}
+        lengths = numpy.array(mapping[numpy.where(right_angles)[0][0]])
+        expected = 0.5*lengths**2.
+        return ((numpy.abs(nonzero_dot-expected) <= tolerance).all() or
+                (numpy.abs(nonzero_dot[::-1]-expected) <= tolerance).all())
+    elif all_length_equal and not one_right_angle:
+        # type 2
+        # p1: 0.5(a, b, -c), p2: 0.5(a, -b, c), p3: 0.5(-a, b, c)
+        # p1, p2, p3 may be flipped and there exists a choice of signs
+        # where
+        #    p1.p2 + p2.p3 + p1.p3 == |p1|**2. == |p2|**2. == |p3|**2.
+        for signs in product(*((1, -1),)*3):
+            if numpy.abs((dot_products*signs).sum() + lenA*lenA) <= tolerance:
+                return True
+        return False
+    else:
+        return False
+
+
+def is_face_centered_orthorhombic_lattice(p1, p2, p3, tolerance=1.e-6):
+    ''' Test if primitive vectors describe a face centered orthorhombic
+    lattice
+
+    Also returns True for vectors describing a face centered cubic
+    lattice
+
+    Numerical errors are tolerated within ``tolerance``
+
+    Parameters
+    ----------
+    p1, p2, p3: array_like
+        primitive vectors
+    tolerance: float
+        tolerance for numerical errors
+
+    Returns
+    -------
+    output : bool
+    '''
+    alpha2, beta2, gamma2 = map(numpy.dot, (p1, p2, p3), (p1, p2, p3))
```

Comment:
A minor thing, but maybe these variables could be named differently as they do not represent angles here.